python/pyspark/sql/session.py

```diff
@@ -952,12 +953,29 @@ def createDataFrame(  # type: ignore[misc]
             schema = [x.encode("utf-8") if not isinstance(x, str) else x for x in schema]
 
         try:
-            import pandas
+            import pandas as pd
 
             has_pandas = True
         except Exception:
             has_pandas = False
-        if has_pandas and isinstance(data, pandas.DataFrame):
+
+        try:
+            import numpy as np
+
+            has_numpy = True
+        except Exception:
+            has_numpy = False
+
+        if has_numpy and isinstance(data, np.ndarray):
+            from pyspark.sql.pandas.utils import require_minimum_pandas_version
+
+            require_minimum_pandas_version()
```

Comment:
nit: If only numpy installed but pandas not installed, will only raised pandas not installed. User may confused why need to install pandas but I just want to using numpy ?

So maybe give a certainly exceptions before here to tell users, numpy type will first convert pandas df in pyspark so pandas installed is required, like:

```python
if not has_pandas:
    // warning or raised friendly exception

from pyspark.sql.pandas.utils import require_minimum_pandas_version
require_minimum_pandas_version()
```

or add a notes before here at least.

Reply:
What you suggested makes sense.

My concern is that 
```py
if not has_pandas:
    // warning or raised friendly exception
```
overlaps with the `require_minimum_pandas_version` check, since the latter checks `has_pandas` as well.

I will leave a note for now and we may follow up with a better approach later. WDYT :)